### DISCUSSION TITLE
Update variable with tabs count

### DISCUSSION THREAD

**Discussion by pwsandoval** [wezterm/wezterm#3181] [Q&A] [answered]
My idea was to change the window_decorations property conditionally, if number of tabs is 1, show the default vlaor "TITLE | RESIZE", but if it's 2 or more change to "RESIZE".

I combine this with hide_tab_bar_if_only_one_tab = true which hides the window bar, which is perfect when viewing the tabs (2 tabs or more, but not when I have 1 tab or open terminal)

I tried to first get the total number of tabs in the current window but it fails (return nil), this is what I got,

```lua
-- To show window bar
local current_window_id = nil
wezterm.on('window-focus-changed', function(window, pane)
   current_window_id = window:window_id()
end)

local function get_tabs_count()
   local current_window = wezterm.mux.get_window(current_window_id)
   local tabs = current_window:tabs()
   return #tabs
end
```

This other work for me 

```lua
local mux = wezterm.mux
local windows = mux.all_windows()[1]:tabs()
print(#windows)
```

But i don't want to use manual value 1 for the window as it wouldn't work if i open more windows, also i get a warning saying:

caused by: cannot get Mux: not running on the mux thread?

**Comment by wez** [comment, marked as answer]
Each thread, and different execution contexts, may run with a fresh lua state.
As such, you cannot share data by storing it directly in global variables.

You can use https://wezfurlong.org/wezterm/config/lua/wezterm/GLOBAL.html for sharing data in that way.

However, because there can be multiple windows, it doesn't make a lot of sense to try to track the window in any kind of global state.

wezterm always passes the current window to the various event callbacks; you should always use the window parameter that is passed to them when you want to operate on the current window.

**Reply by damajor** [reply]
> However, because there can be multiple windows, it doesn't make a lot of sense to try to track the window in any kind of global state.

Then I would say it would be great to have a flag like "is_focused" but related to maxize windows (for eg: "is_maximized").

**Comment by Risinggoblin** [comment]
> My idea was to change the window_decorations property conditionally, if number of tabs is 1, show the default vlaor "TITLE | RESIZE", but if it's 2 or more change to "RESIZE".
> 
> I combine this with hide_tab_bar_if_only_one_tab = true which hides the window bar, which is perfect when viewing the tabs (2 tabs or more, but not when I have 1 tab or open terminal)
> 
> I tried to first get the total number of tabs in the current window but it fails (return nil), this is what I got,
> 
> ```lua
> -- To show window bar
> local current_window_id = nil
> wezterm.on('window-focus-changed', function(window, pane)
>    current_window_id = window:window_id()
> end)
> 
> local function get_tabs_count()
>    local current_window = wezterm.mux.get_window(current_window_id)
>    local tabs = current_window:tabs()
>    return #tabs
> end
> ```
> 
> This other work for me
> 
> ```lua
> local mux = wezterm.mux
> local windows = mux.all_windows()[1]:tabs()
> print(#windows)
> ```
> 
> But i don't want to use manual value 1 for the window as it wouldn't work if i open more windows, also i get a warning saying:
> 
> caused by: cannot get Mux: not running on the mux thread?

I am trying to implement something similar, what did you end up going with for getting tab count and and then dynamically setting window decorations?
